controllers/users.js

```diff
@@ -0,0 +1,107 @@
+const Users = require("../models/userSchema");
+const Joi = require("joi");
+const jwt = require("jsonwebtoken");
+
+const schemaLogin = Joi.object({
+    password: Joi.string().min(10).required(),
+    email: Joi.string().email().required(),
+    subscription: Joi.string().valid("starter", "pro", "business").default("starter"),
+    token: Joi.string().default(null),
+  });
+
+const register = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  const { error } = schemaLogin.validate({email, password });
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
+    }
+
+    try {
+        const user = await Users.findOne({ email }).lean();
+
+        if (user) {
+            return res.status(409).json({ message: 'This email is already taken' });
+        }
+
+        const newUser = new Users({ email });
+        await newUser.setPassword(password);
+        await newUser.save();
+
+      return res.status(201).json(
+        {
+          "user": {
+            "email": email,
+            "subscription": "starter",
+            }
+        });
+    } catch (error) {
+        next(error);
+    }
+};
+
+const login = async (req, res, next) => {
+  
+  const { email, password } = req.body;
+  const user = await Users.findOne({ email })
+  
+  const { error } = schemaLogin.validate({ email, password });
```

Comment:
Najpierw validacja potem zapytanie do bazy danych, ok? Tak we wszystkich kontrolerach gdzie robisz sprawdzenie danych. Najpierw trzeba zobaczyć czy są poprawne, a dopiero potem robić operacje na bazie danych. Czy widzisz co możesz zmienić? :)